spec/requests/publishers/ats_api/v1/vacancies_spec.rb

```diff
@@ -361,31 +112,57 @@
             school_urns: schools.map { |school| school.urn.to_i },
           }
         end
-        let(:vacancy) do
+        let(:vacancy_params) do
           {
-            vacancy: {
-              external_advert_url: source.external_advert_url,
-              expires_at: source.expires_at,
-              job_title: source.job_title,
-              job_advert: source.job_advert,
-              skills_and_experience: source.skills_and_experience,
-              salary: source.salary,
-              visa_sponsorship_available: source.visa_sponsorship_available,
-              external_reference: source.external_reference,
-              publisher_ats_api_client_id: client.id,
-              is_job_share: source.is_job_share,
-              job_roles: source.job_roles,
-              working_patterns: source.working_patterns,
-              contract_type: source.contract_type,
-              phases: source.phases,
-              schools: organisation_ids,
-            },
+            publisher_ats_api_client_id: client.id,
+            external_advert_url: "https://example.com/jobs/123",
+            expires_at: "2026-01-01",
+            job_title: "Teacher of Geography",
+            job_advert: "We're looking for a dedicated Teacher of Geography",
+            salary: "£12,345 to £67,890",
+            visa_sponsorship_available: true,
+            external_reference: "REF1234HYZ",
+            ect_suitable: true,
+            job_roles: %w[teacher],
+            is_job_share: false,
+            working_patterns: %w[full_time],
+            contract_type: "permanent",
+            phases: %w[secondary],
+            publish_on: (Time.zone.today + 1).strftime("%Y-%m-%d"),
+            schools: organisation_ids,
+            subjects: %w[Biology],
+            key_stages: %w[ks1 ks2],
+            starts_on: "Next April",
           }
         end
+        let(:vacancy) { { vacancy: vacancy_params } }
 
-        it "creates a vacancy" do |example|
+        it "creates a vacancy with the given values" do |example|
```

Comment:
Added assertions to test the actual API end-to-end. Before this was just testing the response schema and status (`assert_response_matches_metadata(example.metadata)`)